File: 150. Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        if(tokens.size() == 0)
            return 0;
        stack<int> s;
        for(string str: tokens){
            if(str[0] >= '0' && str[0] <= '9' || str.size() > 1)
                s.push(std::atoi(str.c_str()));
            else
            {
                int b = s.top();
                s.pop();
                int a = s.top();
                s.pop();
                if(str[0] == '+')
                    a += b;
                else if(str[0] == '-')
                    a -= b;
                else if(str[0] == '*')
                    a *= b;
                else if(str[0] == '/')
                    a /= b;
                s.push(a);
            }
        }
        return s.top();
    }
};

File: 150. Evaluate Reverse Polish Notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        if(tokens.length == 0)
            return 0;
        Stack<Integer> s = new Stack<Integer>();
        for(String str: tokens){
            if(str.equals("+"))
                s.push(s.pop() + s.pop());
            else if(str.equals("*"))
                s.push(s.pop() * s.pop());
            else if(str.equals("-")){
                int b = s.pop();
                int a = s.pop();
                s.push(a - b);
            }
            else if(str.equals("/")){
                int b = s.pop();
                int a = s.pop();
                s.push(a / b);
            }
            else
                s.push(Integer.parseInt(str));
        }
        return s.peek();
    }
}